lib/controller.dart

```diff
@@ -117,20 +118,102 @@ void init() async {
 void onLoginCompleted() async {
   view.showLoading();
 
-  await loadFirebaseData();
-  if (_config.data_paths == null || _config.tabs == null) {
-    view.hideLoading();
-    return;
+  fb.listenToConfig((DocumentSnapshot documentSnapshot) async {
+    _configRaw = documentSnapshot.data();
+    _config = model.Config.fromData(_configRaw);
+
+    await loadGeoMapsData();
+    _handleInteractionsChanges();
+    _handleSurveyStatusChanges();
+    _handleMessageStatusChanges();
+  }, (error) {
+    view.showAlert(UNABLE_TO_PARSE_CONFIG_ERROR_MSG);
+    logger.error(error.toString());
+  });
+}
+
+void _handleInteractionsChanges() {
+  if (_config.data_paths['interactions'] == null) return;
+  var interactionsPath = _config.data_paths['interactions']['data'];
+  if (interactionsPath != null) {
+    fb.listenToInteractions(interactionsPath, (QuerySnapshot querySnapshot) {
+      var interactionsMap = Map<String, Map<String, dynamic>>();
+      querySnapshot.forEach((doc) {
+        interactionsMap[doc.id] = doc.data();
+      });
+      _allInteractions = interactionsMap;
+      _handleDataChanges();
+    }, (error) {
+      view.showAlert(UNABLE_TO_FETCH_INTERACTIONS_ERROR_MSG);
+      logger.error(error.toString());
+    });
+  }
+}
+
+void _handleSurveyStatusChanges() {
```

Comment:
these look very similar with each other - I think they fall under the issue of handling the different data types together, so let's keep these as they are for now and generalise next?

Reply:
Yep will generalise it along with storing the data collections in the controller